src/NextGen/Framework/Blocks/BlockCollection.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Give\NextGen\Framework\Blocks;
+
+use ArrayObject;
+use Give\Framework\Support\Contracts\Arrayable;
+
+class BlockCollection extends ArrayObject implements Arrayable
```

Comment:
We're extending `ArrayObject`, but it doesn't actually look like we're using it as such.

Reply:
`ArrayObject` allows for the iteration without having to write the offset methods.

Reply:
Yes, but I don't see where we're actually using `$this['foo']` or something like. It looks like the `blocks` property is the underlying array being accessed, which I don't think is how `ArrayObject` works — it has an implicit store.

Reply:
@JasonTheAdams It looks like this was lost in a refactor. Originally, we were using the iterable property of the parent ArrayObject, but we since started using a custom $blocks property, which loses the ability to iterate the ArrayObject items.

@jonwaldstein we should probably discuss this in a call.

Reply:
Oh yeah, I needed to change how `toArray()` works for the serialization to work properly.  Originally Kyle was using the ArrayObject's `getArrayCopy` method which would have been nice but it wasn't in the format we needed.

Reply:
@jonwaldstein @JasonTheAdams on the next call we should probably discuss the Block Model and Collection to get on the same page.

Reply:
sure thing

Reply:
Call Notes:

Need to remove the `ArrayObject` base class as it is no longer necessary.